vividus/src/test/java/org/vividus/runner/BddScenariosCounterTests.java

```diff
@@ -0,0 +1,155 @@
+
+        Description root = Description.createSuiteDescription(ROOT);
+        Description beforeStories = Description.createTestDescription(Object.class, BEFORE);
+        Description story = Description.createSuiteDescription(STORY);
+        Description beforeStory = Description.createTestDescription(Object.class, BEFORE);
+        Description scenario = Description.createSuiteDescription(SCENARIO);
+
+        root.addChild(beforeStories);
+        root.addChild(story);
+        story.addChild(beforeStory);
+        story.addChild(scenario);
+
+        when(runner.getDescription()).thenReturn(root);
+
+        BddScenariosCounter.main(new String[0]);
+
+        assertTrue(getOutput().contains(String.format(FORMATTER, 1, STORIES)));
+        assertTrue(getOutput().contains(String.format(FORMATTER, 1, SCENARIOS)));
+        assertTrue(getOutput().contains(String.format(FORMATTER, 1, SCENARIOS_WITH_EXAMPLES)));
+        verify(properties).put(RESOURCE_LOCATION, DEFAULT_STORY_LOCATION);
+    }
+
+    @Test
+    public void testMultipleChildDescriptions() throws Exception
+    {
+        stubPropertiesAndRunnerInvocations();
+
+        Description root = Description.createSuiteDescription(ROOT);
+        Description story = Description.createSuiteDescription(STORY);
+        Description scenario = Description.createSuiteDescription(SCENARIO);
+        Description scenarioWithoutExamples = Description.createSuiteDescription(SCENARIO);
+        Description example = Description.createSuiteDescription(EXAMPLE);
+
+        root.addChild(story);
+        story.addChild(scenario);
+        story.addChild(scenarioWithoutExamples);
+        scenario.addChild(example);
+        scenario.addChild(example);
+
+        when(runner.getDescription()).thenReturn(root);
+
+        BddScenariosCounter.main(new String[0]);
+
+        assertTrue(getOutput().contains(String.format(FORMATTER, 1, STORIES)));
+        assertTrue(getOutput().contains(String.format(FORMATTER, 2, SCENARIOS)));
+        assertTrue(getOutput().contains(String.format(FORMATTER, 3, SCENARIOS_WITH_EXAMPLES)));
+    }
+
+    @Test
+    public void testDirectoryOptionIsPresent() throws Exception
+    {
+        stubPropertiesAndRunnerInvocations();
+        when(runner.getDescription()).thenReturn(Description.createSuiteDescription(ROOT));
+        BddScenariosCounter.main(new String[] {"--dir", DIR_VALUE});
+        verify(properties).put(RESOURCE_LOCATION, DIR_VALUE);
+    }
+
+    @Test
+    public void testUnknownOptionIsPresent()
+    {
+        assertThrows(UnrecognizedOptionException.class, () ->
+                BddScenariosCounter.main(new String[] { "--any", DIR_VALUE}));
+    }
+
+    @Test
+    public void testHelpOptionIsPresent() throws Exception
+    {
+        BddScenariosCounter.main(new String[] {"--help"});
+        assertTrue(getOutput().contains("usage: BddScenariosCounter"));
+    }
+
+    private void stubPropertiesAndRunnerInvocations() throws Exception
```

Comment:
`...And...` in the method name usually means that the method does more than one thing, but more than one thing is a violation of [Single responsibility principle](https://stackoverflow.com/questions/46541197/does-the-single-responsibility-principle-work-in-oop/46556381#46556381).  So what can be done here:
- `when(BeanFactory...` can't be used without `PowerMockito.mockStatic(BeanFactory.class);` so they can be grouped and placed into a new method
- mock `runner` is not used in all tests, so it makes sense to move its creation and `whenNew(JUnitReportingRunner...` to another method and this method should return just created mock:
```java
private JUnitReportingRunner mockJUnitReportingRunnerInstantiation()
{
    JUnitReportingRunner runner = mock(JUnitReportingRunner.class);
    whenNew(JUnitReportingRunner.class).withArguments(StoriesRunner.class).thenReturn(runner);
    return runner;
}
```
```